psh/echo/echo.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Phoenix-RTOS
+ *
+ * echo - display a line of text
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "../psh.h"
+
+
+void psh_echoinfo(void)
+{
+	printf("display a line of text");
+}
+
+
+static void psh_echo_help(const char *prog)
+{
+	printf("Usage: %s [options] [string]\n", prog);
+	printf("  -h:  shows this help message\n");
+}
+
+
+size_t psh_echoPrintVar(const char *var)
+{
+	/* Right now only $? is supported */
+	if (*var != '?')
+		return 0;
+
+	printf("%d", psh_common.exitStatus);
+
+	return 1; /* Eaten variable length */
+}
+
+
+int psh_echo(int argc, char **argv)
+{
+	int c, i;
+	size_t j;
+
+	while ((c = getopt(argc, argv, "h")) != -1) {
+		switch (c) {
+			case 'h':
+			default:
+				psh_echo_help(argv[0]);
+				return EOK;
+		}
+	}
+
+	for (i = 1; i < argc; ++i) {
+		if (i != 1)
+			putchar(' ');
+
+		for (j = 0; argv[i][j] != '\0'; ++j) {
+			if (argv[i][j] == '$') {
+				j += psh_echoPrintVar(&argv[i][j + 1]);
+			}
+			else if (argv[i][j] != '"') { /* Primitive - just eat "" */
+				putchar(argv[i][j]);
+			}
```

Comment:
I thought that our convention says not to use brackets, when there is only one line in an `if` statement.
```suggestion
			if (argv[i][j] == '$')
				j += psh_echoPrintVar(&argv[i][j + 1]);
			else if (argv[i][j] != '"') /* Primitive - just eat "" */
				putchar(argv[i][j]);
```

Reply:
Yeah, we're sailing away from that due to MISRA. We still need to update our coding conventions, though

Reply:
Ok, good to know. I'll try to follow this MISRA compilant style in the future.